packages/create-fuels/src/cli.ts

```diff
@@ -36,20 +37,36 @@ function writeEnvFile(envFilePath: string) {
   writeFileSync(envFilePath, newFileContents);
 }
 
+const doesTemplateExist = (templateName: Template): boolean => templates.has(templateName);
+
 export const runScaffoldCli = async ({
   program,
-  templateName = 'nextjs',
+  explicitTemplateName, // Only to be used for testing. The actual template name is passed in as a CLI arg.
   args = process.argv,
+  checkIfTemplateExists = true,
 }: {
   program: Command;
   args: string[];
-  templateName: string;
+  explicitTemplateName?: string;
+  checkIfTemplateExists?: boolean;
 }) => {
   program.parse(args);
+  const opts = program.opts<ProgramOptions>();
+
+  const templateOfChoice = (explicitTemplateName ?? opts.template ?? 'nextjs') as Template;
+
+  if (!doesTemplateExist(templateOfChoice) && checkIfTemplateExists) {
+    error(`Template '${templateOfChoice}' does not exist.`);
+    log();
+    log('Available templates:');
+    for (const template of templates) {
+      log(`  - ${template}`);
+    }
+    process.exit(1);
+  }
```

Comment:
Instead of `checkIfTemplateExists` for testing purposes, could we not mock this function in the test?

Reply:
https://github.com/FuelLabs/fuels-ts/pull/2851/commits/b6c15e941032840637832d82f29eaf6b7b25d5dc